dev/benchmarks/test_apps/stocks/lib/main.dart

```diff
@@ -65,49 +65,140 @@ class StocksAppState extends State<StocksApp> {
     assert(() {
-      debugPaintSizeEnabled = _configuration.debugShowSizes;
-      debugPaintBaselinesEnabled = _configuration.debugShowBaselines;
-      debugPaintLayerBordersEnabled = _configuration.debugShowLayers;
-      debugPaintPointersEnabled = _configuration.debugShowPointers;
-      debugRepaintRainbowEnabled = _configuration.debugShowRainbow;
+      debugPaintSizeEnabled = configuration.debugShowSizes;
+      debugPaintBaselinesEnabled = configuration.debugShowBaselines;
+      debugPaintLayerBordersEnabled = configuration.debugShowLayers;
+      debugPaintPointersEnabled = configuration.debugShowPointers;
+      debugRepaintRainbowEnabled = configuration.debugShowRainbow;
       return true;
     }());
-    return MaterialApp(
+    return MaterialApp.router(
       title: 'Stocks',
-      theme: theme,
+      theme: geThemeFromConfiguration(configuration),
       localizationsDelegates: StockStrings.localizationsDelegates,
       supportedLocales: StockStrings.supportedLocales,
-      debugShowMaterialGrid: _configuration.debugShowGrid,
-      showPerformanceOverlay: _configuration.showPerformanceOverlay,
-      showSemanticsDebugger: _configuration.showSemanticsDebugger,
-      routes: <String, WidgetBuilder>{
-         '/':         (BuildContext context) => StockHome(stocks, _configuration, configurationUpdater),
-         '/settings': (BuildContext context) => StockSettings(_configuration, configurationUpdater),
-      },
-      onGenerateRoute: _getRoute,
+      debugShowMaterialGrid: configuration.debugShowGrid,
+      showPerformanceOverlay: configuration.showPerformanceOverlay,
+      showSemanticsDebugger: configuration.showSemanticsDebugger,
+      routeInformationParser: _routeInformationParser,
+      routerDelegate: _routerDelegate,
     );
   }
 }
 
-void main() {
-  runApp(StocksApp());
+class StockRouteInformationParser extends RouteInformationParser<StockRoutePath> {
+  @override
+  Future<StockRoutePath> parseRouteInformation(RouteInformation routeInformation) {
+    final Uri url = Uri.parse(routeInformation.location);
+    if (url.path == _kSettingsPageLocation) {
+      return SynchronousFuture<StockRoutePath>(const StockSettingsPath());
+    }
+
+    if (url.path == _kStockPageLocation) {
+      final String symbol = url.queryParameters['symbol'];
+      if (symbol != null && symbol.isNotEmpty)
+        return SynchronousFuture<StockRoutePath>(StockSymbolPath(symbol));
+    }
+
+    return SynchronousFuture<StockRoutePath>(const StockHomePath());
+  }
+
+  @override
+  RouteInformation restoreRouteInformation(StockRoutePath configuration) {
+    if (configuration is StockSettingsPath)
+      return const RouteInformation(location: _kSettingsPageLocation);
+    if (configuration is StockHomePath)
+      return const RouteInformation(location: _kHomePageLocation);
+    if (configuration is StockSymbolPath)
+      return RouteInformation(location: '$_kStockPageLocation?symbol=${configuration.symbol}');
```

Comment:
the symbol needs escaping, otherwise you have (in principle, anyway) an injection vulnerability.

Reply:
(i'm assuming the constants don't need escaping because they're designed to not have anything outside of the local appropriate alphabet)